### PR TITLE
Cannot add a custom elasticsearch field mapper to `Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\Resolver\CompositeResolver`.

### LINKED ISSUE #24350: Cannot add a custom elasticsearch field mapper to `Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\Resolver\CompositeResolver`
### Description (*)
I'm adding a custom filter to the layered navigation. My filter is not a product attribute. 
this filter has a request key that does not match the field I store in elasticsearch (similar to how the price works. Price request key is `price` but in elasticsearch is stored as `price_{customer_group}_{website_id}`).  
I need to map this request key to the elasticsearch field. 
I've found that this is done via `Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\Resolver\CompositeResolver` which goes through a list of mappers so I tried to add my own.

### Preconditions (*)
1. Magento 2.3.2

### Steps to reproduce (*)
1. Create a custom module that adds one custom filter in the layered navigation
2. Create one field resolver class that implements `Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\ResolverInterface`

```
namespace [Vendor]\[Module]\Model;

class CustomFieldMapper implements \Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\ResolverInterface
{
    public function getFieldName(AttributeAdapter $attribute, $context = []): ?string
    {
         return $attribute->getAttributeCode() == 'some_value' ? 'some_other_value' : null;
    }
}
```
3. Attach the new mapper to the composite mapper

```
    <type name="Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\Resolver\CompositeResolver">
        <arguments>
            <argument name="items" xsi:type="array">
                <item name="custom_mapper" xsi:type="object">[Vendor]\[Module]\Model\CustomFieldMapper</item>
            </argument>
        </arguments>
    </type>
```

### Expected result (*)

1. My custom mapper should get called
2. The call to  `Magento\Elasticsearch\Model\Adapter\FieldMapper\Product\FieldProvider\FieldName\Resolver\CompositeResolver::getFieldName($attribute)` where `$attribute->getAttributeCode()` returns `some_value` should return `some_other_value`

### Actual result (*)
1. The call from above returns `some_value` because my custom field mapper never gets called. This happens because the default mapper always returns something and my mapper is lower that the default mapper in the list of mappers. 

### Additional info:
My mapper gets attached to the conpositeResolver class properly.   
If I put my mapper higher in the list (editing a core file because I cannot do it via a module) everything works as expected.
I know I can go around it using a plugin for the `ResolverInterface` method, but that kind of defeats the purpose of using composite resolvers.

### Suggestion(s):
 - have a sort order for the mappers
 - (or) ensure that `default` is not part of the list of mappers and it is always executued last.

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/24350

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/24350

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Elasticsearch/Elasticsearch5/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/Resolver/CompositeResolver.php [hunk 1]
```diff
@@ -33,6 +33,10 @@ public function __construct(array $items)
             }
         }
         $this->items = $items;
+        unset($this->items['default']);
```
Comment (zakdma) [inline app/code/Magento/Elasticsearch/Elasticsearch5/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/Resolver/CompositeResolver.php]: Please consider to use sortOrder attribute of array elements that can be specified in DI
This hardcoded 'default' key looks unreliable for me

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Elasticsearch/Elasticsearch5/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/Resolver/CompositeResolver.php (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Elasticsearch/Elasticsearch5/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/Resolver/CompositeResolver.php]: done

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32742`.

**Comment by dmitriyprime** [comment]
:heavy_check_mark:  QA passed

**Pull request merged**